modules/govuk_scripts/templates/govuk_node_list_aws.erb

```diff
@@ -42,6 +42,25 @@ def ec2_nodes(stackname, nodeclass=None):
 
     return(hosts)
 
+def ec2_nodes_list(stackname):
+    client = boto3.client('ec2', region_name=region)
+
+    response = client.describe_instances(
+                   Filters=[
+                          {'Name': 'tag:aws_stackname', 'Values': [stackname]},
+                          {'Name': 'instance-state-name','Values': ['running']}
+                   ]
+               )
+
+    nodes = response['Reservations']
+
+    hosts = []
+    for instance in nodes:
+        node_name=list(filter(lambda t: t['Key'] == 'aws_hostname', instance['Instances'][0]['Tags']))[0]['Value']
+        hosts.append((instance['Instances'][0]['PrivateDnsName']+":"+node_name))
+
+    return(hosts)
```

Comment:
Using a list comprehension would avoid mutating the hosts variable. Also, you can use `next` to get the first element of a generator without converting it to a list. Something like this...

```python
    return [
        "%s:%s" % (
            instance['Instances'][0]['PrivateDnsName'],
            next(
                filter(
                    lambda t: t['Key'] == 'aws_hostname',
                    instance['Instances'][0]['Tags']
                )
            )['Value']
        )
        for instance in nodes
    ]
```

Reply:
nice, I need to modify and test it! I followed too much what was done `ec2_nodes` function